docs/mathematics/gcd-euclidean-algorithm.mdx

```diff
@@ -0,0 +1,125 @@
+---
+title: GCD using Euclidean algorithm
+slug: /mathematics/gcd-euclidean-algorithm
+---
+
+## Introduction
+
+The GCD of two numbers is the largest integer that divides each of the integers such that their remainder is zero. The Euclidean algorithm is an efficient way for finding the GCD of `2` numbers. It is named after the Greek mathematician Euclid
+
+## Explanation
+
+Consider two numbers `a = 333` and `b = 84`
+
+`333 = 3*84 + 81`   `0 < 81 < 84`
+
+`84 = 1*81 + 3`     `0 < 3 < 81`
+
+`81 = 27*3 + 0`
+
+Since 3 is the last nonzero remainder, the euclidean algorithm tells us that GCD(a,b) = 3.
+
+## Algorithm
+
+- input `a` and `b`
+- repeat steps until a mob b is not 0 
+    - `r = a mod b`
+    - `a = b`
+    - `b = r`
+- return `b`
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+int gcd(int a,int b)
+{ // base case
+  if (a==0)
+     return b;
+  // recursive case
+  else
+     return gcd(b%a,a);
+}
+
+// initializing a and b.
+// Here a is always the smaller number and
+// b is always the bigger number between the two.
+
+int main()
+{
+    cout<<"Enter a and b for finding GCD"<<endl;
```

Comment:
Write proper statement
**Enter a and b for finding GCD** is vague statement
Write something like this, Enter two numbers........